packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -57,15 +138,24 @@ export class Wallet {
    * @returns A promise that resolves to a collection of ObjectiveResult.
    */
   public async approveObjectives(objectiveIds: string[]): Promise<ObjectiveResult[]> {
-    const {objectives, messages} = await this._engine.approveObjectives(objectiveIds);
-    return Promise.all(
-      objectives.map(async o => ({
-        objectiveId: o.objectiveId,
-        currentStatus: o.status,
-        channelId: o.data.targetChannelId,
-        done: this.ensureObjective(o, messages),
-      }))
+    const {objectives, messages, chainRequests} = await this._engine.approveObjectives(
+      objectiveIds
     );
+
+    const results = objectives.map(async o => ({
+      objectiveId: o.objectiveId,
+      currentStatus: o.status,
+      channelId: o.data.targetChannelId,
+      done: this.createObjectiveDoneResult(o),
+    }));
+
+    // TODO: ApproveObjective should probably just return a MultipleChannelOuput
```

Comment:
Why is there an `approveObjectives` method, instead of just a `approveObjective` method?

Reply:
What's the advantage of having `approveObjective` instead of `approveObjectives` ?